spec/models/user_spec.rb

```diff
@@ -22,3 +22,108 @@
     expect(user.next_slot_in_queue).to eq 1
   end
 end
+
+describe '#remove_from_queue!' do
+  let(:user) { Fabricate :user }
+
+  it 'deletes the input queue item, if it belongs to the current user' do
+    item_to_delete = Fabricate :queue_item, user: user
+    user.remove_from_queue!(item_to_delete)
+    expect(QueueItem.all.count).to eq 0
+  end
+
+  it 'does not delete the queue item, if it does not belong to current user' do
+    item_to_delete = Fabricate :queue_item
+    user.remove_from_queue!(item_to_delete)
+    expect(QueueItem.all.count).to eq 1
+  end
+end
+
+describe '#has_in_queue?' do
+  let(:user) { Fabricate :user }
+  let(:south_park) { Fabricate :video, title: 'South Park', tagline: 'hahahaha so so funny!' }
+
+  it "returns true if the given video is in the user's queue" do
+    Fabricate :queue_item, video: south_park, user: user
+    expect(user.has_in_queue? south_park).to be true
+  end
+
+  it "returns false if the given video is not in the user's queue" do
+    expect(user.has_in_queue? south_park).to be nil
+  end
+end
+
+describe '#update_queue!' do
+  let(:user) { Fabricate :user }
+  let!(:queue) { render_queue(user, 3) }
+
+  it 'does not update the queue if any item submitted for update is not associated with the user' do
+    invalid_queue = queue << Fabricate(:queue_item)
+    invalid_queue_params = render_queue_params invalid_queue
+    user.update_queue! invalid_queue_params
+    queue.pop
+    expect(user.queue_items).to match_array queue
```

Comment:
I know it saves a lot of code, but I am not a big fan of the `queue` abstraction. There is a lot of magic happening. Your tests should be explicit, so that it is very obvious if your test is wrong, rather than the implementation.